pkg/shp/flags/build.go

```diff
@@ -76,4 +93,24 @@ func SanitizeBuildSpec(b *buildv1alpha1.BuildSpec) {
 	if b.Dockerfile != nil && *b.Dockerfile == "" {
 		b.Dockerfile = nil
 	}
+	if b.Output.Credentials != nil && b.Output.Credentials.Name == "" {
+		b.Output.Credentials = nil
+	}
+	if b.Retention != nil {
+		if b.Retention.FailedLimit != nil && *b.Retention.FailedLimit == 65535 {
```

Comment:
nit: would be good to have the `FailedLimit` value as a constant imported from the build controller repository directly.